awx/ui_next/src/components/CodeMirrorInput/VariablesField.jsx

```diff
@@ -1,16 +1,16 @@
 import React, { useState } from 'react';
 import { string, bool } from 'prop-types';
-import { Field } from 'formik';
+import { Field, useFormikContext } from 'formik';
 import { Split, SplitItem } from '@patternfly/react-core';
+import { yamlToJson, jsonToYaml, isJson } from '@util/yaml';
 import CodeMirrorInput from './CodeMirrorInput';
 import YamlJsonToggle from './YamlJsonToggle';
-import { yamlToJson, jsonToYaml } from '../../util/yaml';
-
-const YAML_MODE = 'yaml';
+import { JSON_MODE, YAML_MODE } from './constants';
 
 function VariablesField({ id, name, label, readOnly }) {
-  // TODO: detect initial mode
-  const [mode, setMode] = useState(YAML_MODE);
+  const { values } = useFormikContext();
+  const value = values[name];
+  const [mode, setMode] = useState(isJson(value) ? JSON_MODE : YAML_MODE);
```

Comment:
I like this. In looking at how these utils work and how this component makes use of them, I think the assumption we're making is that if the value is a string and *not* json, just assume the API sent valid yaml. 

To save extra vars on a job template, the string must either be yaml or json. I presume the other `*variables` fields in the API work similarly, so this makes sense to me. :+1: